Analysis/Tasks/PWGCF/FemtoDream/include/FemtoDream/FemtoDreamMath.h

```diff
@@ -0,0 +1,99 @@
+/// \author Valentina Mantovani Sarti, TU München, valentina.mantovani-sarti@ph.tum.de
+
+#ifndef ANALYSIS_TASKS_PWGCF_FEMTODREAM_FEMTODREAMMATH_H_
+#define ANALYSIS_TASKS_PWGCF_FEMTODREAM_FEMTODREAMMATH_H_
+
+#include "Math/Vector4D.h"
+#include "Math/Boost.h"
+#include "TLorentzVector.h"
+#include "TMath.h"
+
+#include <iostream>
+
+namespace o2::analysis
+{
+namespace femtoDream
+{
+
+/// \class FemtoDreamMath
+/// \brief Container for math calculations of quantities related to pairs
+
+class FemtoDreamMath
+{
+ public:
+
+  template <typename T>
+  static float getkstar(const T& part1, const float mass1, const T& part2, const float mass2)
+  {
+    ROOT::Math::PtEtaPhiMVector vecpart1(part1.pt(), part1.eta(), part1.phi(), mass1);
+    ROOT::Math::PtEtaPhiMVector vecpart2(part2.pt(), part2.eta(), part2.phi(), mass2);
+
+    ROOT::Math::PtEtaPhiMVector trackSum = vecpart1 + vecpart2;
+
+    float beta = trackSum.Beta();
+    float betax = beta * std::cos(trackSum.Phi()) * std::sin(trackSum.Theta());
+    float betay = beta * std::sin(trackSum.Phi()) * std::sin(trackSum.Theta());
+    float betaz = beta * std::cos(trackSum.Theta());
+
+    float px1 = vecpart1.pt() * std::cos(vecpart1.Phi());
+    float py1 = vecpart1.pt() * std::sin(vecpart1.Phi());
+    float pz1 = vecpart1.pt() * std::sinh(vecpart1.Eta());
+    float px2 = vecpart2.pt() * std::cos(vecpart2.Phi());
+    float py2 = vecpart2.pt() * std::sin(vecpart2.Phi());
+    float pz2 = vecpart2.pt() * std::sinh(vecpart2.Eta());
+
+    ROOT::Math::PxPyPzMVector PartOneCMS(px1, py1, pz1, vecpart1.M());
+    ROOT::Math::PxPyPzMVector PartTwoCMS(px2, py2, pz2, vecpart2.M());
+
+    ROOT::Math::Boost boostPRF = ROOT::Math::Boost(-betax, -betay, -betaz);
+    PartOneCMS = boostPRF(PartOneCMS);
+    PartTwoCMS = boostPRF(PartTwoCMS);
+
+    ROOT::Math::PxPyPzMVector trackRelK = PartOneCMS - PartTwoCMS;
+    return 0.5 * trackRelK.P();
+  }
+
+  template <typename T>
+  static float getkT(const T& part1, const float mass1, const T& part2, const float mass2)
+  {
+    ROOT::Math::PtEtaPhiMVector vecpart1(part1.pt(), part1.eta(), part1.phi(), mass1);
+    ROOT::Math::PtEtaPhiMVector vecpart2(part2.pt(), part2.eta(), part2.phi(), mass2);
+
+    ROOT::Math::PtEtaPhiMVector trackSum = vecpart1 + vecpart2;
+
+    float trackRelkT = 0.;
+    trackRelkT = 0.5 * trackSum.Pt();
+    return trackRelkT;
+  }
+
+  template <typename T>
+  static float getmT(const T& part1, const float mass1, const T& part2, const float mass2)
+  {
+    ROOT::Math::PtEtaPhiMVector vecpart1(part1.pt(), part1.eta(), part1.phi(), mass1);
+    ROOT::Math::PtEtaPhiMVector vecpart2(part2.pt(), part2.eta(), part2.phi(), mass2);
+
+    ROOT::Math::PtEtaPhiMVector trackSum = vecpart1 + vecpart2;
+    float relmT = 0.;
+    float trackRelkT = 0.5 * trackSum.Pt();
+    float averageMass = 0.5 * (mass1+ mass2);
+    relmT = TMath::Sqrt(pow(trackRelkT, 2) + pow(averageMass, 2));
+    return relmT;
```

Comment:
same here
return std::sqrt(std::pow(getkT(..), 2) + pow(0.5 * (m1+m2), 2));

(also note that std::sqrt etc. is always better than the root TMath functions)